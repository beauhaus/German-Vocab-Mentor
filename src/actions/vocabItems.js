import uuid from 'uuid';



// ADD VOCAB
export const addVocabItem = ({
    word_de = '',
    word_en = '',
    example_de = '',
    example_en = '',
    isShowing = '',
    note = '',
    createdAt = 0 } = {}
) => ({
    type: 'ADD_VOCAB_ITEM',
    vocab: {
        "id": uuid(),
        word_de,
        word_en,
        example_de,
        example_en,
        isShowing,
        note,
        createdAt
    }
})

// REMOVE VOCAB

export const removeVocabItem = ({ id } = {}) => ({
    type: 'REMOVE_VOCAB_ITEM',
    id
});

// EDIT VOCAB
export const editVocabItem = (id, updates) => ({
    type: 'EDIT_VOCAB_ITEM',
    id,
    updates
});