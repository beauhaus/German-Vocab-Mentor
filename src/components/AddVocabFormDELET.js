import React, {Component} from 'react'
import VocabForm from './VocabForm';

class AddVocabPage extends Component {
    render() {
        return (
            <div>
            <h1>AddVocabPage</h1>
            <VocabForm/>
            </div>
        )
    }

}

export default AddVocabPage;