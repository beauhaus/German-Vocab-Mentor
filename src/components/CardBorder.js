import React from 'react';


const CardBorder = (props) => (
    <svg className="card-border card-border-de" viewBox="0 0 405 405" preserveAspectRatio="none">
        <linearGradient id="card-border-grad-de" x1="200" x2="200" y2="400" gradientUnits="userSpaceOnUse">
            <stop offset=".1302" />
            <stop offset=".3785" stopColor="#69001e" />
            <stop offset=".4992" stopColor="#97002b" />
            <stop offset=".7395" stopColor="#d27d0d" />
            <stop offset=".8502" stopColor="#ebb100" />
        </linearGradient>
        <linearGradient id="card-border-grad-en" x1="200" x2="200" y2="400" gradientUnits="userSpaceOnUse">
            <stop offset=".065" stopColor="#940000" />
            <stop offset=".104" stopColor="#971111" />
            <stop offset=".378" stopColor="#ab8585" />
            <stop offset=".499" stopColor="#b3b3b3" />
            <stop offset=".768" stopColor="#3434a8" />
            <stop offset=".89" stopColor="#0000a3" />
        </linearGradient>
        <rect width="400" height="395" x="2.5" y="2.5" stroke={props.langSwitch?"url(#card-border-grad-de)":"url(#card-border-grad-en)"} />
    </svg>
);

module.exports = CardBorder;