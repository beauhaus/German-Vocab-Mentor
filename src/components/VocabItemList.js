import React from 'react';
import {connect} from 'react-redux';
import VocabListItem from './VocabListItem';
import selectVocabItems from '../reduxUtils/selectors/vocabSelectors';

const VocabItemList = (props) =>{
    return (
    <div>
       <h1>VocabItemList</h1>
       <h2><em>{props.vocabItems.length}</em></h2>
       <h2>{props.vocabItems.map((vocabItem) => {
           return <VocabListItem key={vocabItem.id} {...vocabItem}/>
        })}
        </h2>
    </div>
)}
// http://localhost:8080/VocabItemDashboard
const MapStateToProps =(state)=> {
    return {
        vocabItems: selectVocabItems(state.vocabItems, state.filters)
    }
} 

export default connect(MapStateToProps)(VocabItemList)