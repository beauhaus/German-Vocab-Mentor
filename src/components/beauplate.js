import React, { Component } from 'react';
import { connect } from 'react-redux';
import VocabListItem from './VocabListItem';
import selectVocabItems from '../reduxUtils/selectors/vocabSelectors';
import FiftyButton from './Modals/FiftyButton';
import FiftyModal from './Modals/FiftyModal';

export class beauplate extends Component {
    state = {
        fiftyModalSample: null
    }
    fiftyModalClickHandler=(e)=> {
        e.preventDefault();
        const val = e.target.value;
        console.log("val: ", val)
        this.setState(() =>({
            fiftyModalSample: val
        }))
    }
    fiftyModalClear=()=> {        
        this.setState(() =>({
            fiftyModalSample: null
        }))
    }
    render() {
        const {vocabItems} = this.props;
        return (
            <div>
                <p>beauplate</p>
                {vocabItems.map((item, idx, arr)=>{
                    return(
                        (idx%50 === 0)?
                        <FiftyButton key={idx} idx={idx} fiftyModalClickHandler={this.fiftyModalClickHandler}/>:
                     '');
                })}

                <FiftyModal 
                fiftyModalSample={this.state.fiftyModalSample}
                fiftyModalClear={this.fiftyModalClear}
                />
                {/*
                {this.props.vocabItems.length === 0 ? (
                    <p>Sorry. length is zero.</p>
                ) :
                    (<h2>{this.props.vocabItems.map((vocabItem) => {
                        return <VocabListItem key={vocabItem.id} {...vocabItem} />
                    })}
                    </h2>)
                }
            */}
            </div>
        )
    }
}

const MapStateToProps = (state) => {
    return {
        vocabItems: selectVocabItems(state.vocabItems, state.filters)
    }
}
export default connect(MapStateToProps)(beauplate)

// export default beauplate;