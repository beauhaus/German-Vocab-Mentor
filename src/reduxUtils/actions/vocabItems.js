import uuid from 'uuid';

export const addVocabItem = ({ word_de = '', word_en = '', note = '', difficulty = 10, isShowing = true, example_de = '', example_en = '', createdAt = 0 } = {}) => ({
    type: 'ADD_VOCAB_ITEM',
    vocabItem: {
        _id: uuid(),
        word_de,
        word_en,
        note,
        example_de,
        example_en,
        difficulty,
        createdAt,
        isShowing
    }
})

export const removeVocabItem = ({ _id } = {}) => ({
    type: 'REMOVE_VOCAB_ITEM',
    _id
})

export const editVocabItem = (_id, updates) => ({
    type: 'EDIT_VOCAB_ITEM',
    _id,
    updates
})