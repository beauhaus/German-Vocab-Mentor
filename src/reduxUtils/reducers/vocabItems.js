const vocabItemsReducerDefaultState = [];

const vocabItemsReducer = (state = vocabItemsReducerDefaultState, action) => {
    switch (action.type) {
        case 'ADD_VOCAB_ITEM':
            return [...state, action.vocabItem]
        case 'REMOVE_VOCAB_ITEM':
            return state.filter(({ _id }) => _id !== action._id)
        case 'EDIT_VOCAB_ITEM':
            return state.map((vocabItem) => {
                if (vocabItem._id === action._id) {
                    return {
                        ...vocabItem,
                        ...action.updates
                    }
                } else {
                    return vocabItem;
                }
            })
        default:
            return state;
    }
}

export default vocabItemsReducer;