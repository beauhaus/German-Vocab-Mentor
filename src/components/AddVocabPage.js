import React from 'react';
import { Component } from 'react'
import { connect } from 'react-redux';
import styled from 'styled-components';
import VocabForm from './VocabForm';
import { addVocabItem } from '../reduxUtils/actions/vocabItems';

const StyledAddVocabPage = styled.div`
    width: 100vw;
    height: 100vh;
    grid-column: 1;
    grid-row: 2/-1;
    background: #3b6064;
    & label, input, textarea {
        margin: 0.5vh 0.5vw;
        width: 40%;
    }
`;

export class AddVocabPage extends Component {
    onSubmit = (vocabItem) => {
        this.props.addVocabItem(vocabItem);
        this.props.history.push('/vocabItemDashboard')
    }
    render() {
        return (
            <StyledAddVocabPage className="add-vocab-page">
                <VocabForm onSubmit={this.onSubmit} />
            </StyledAddVocabPage>
        )
    }
}

const mapDispatchToProps = (dispatch) => ({
    addVocabItem: (vocabItem) => dispatch(addVocabItem(vocabItem))
})

export default connect(undefined, mapDispatchToProps)(AddVocabPage);