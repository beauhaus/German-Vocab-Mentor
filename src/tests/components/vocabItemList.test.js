import React from 'react';
import {shallow} from 'enzyme';

import {VocabItemList} from '../../components/VocabItemList';
import vocabItems from '../fixtures/vocabItems';

test('should render vocabItems list with vocab items', () => {
    const wrapper = shallow(<VocabItemList vocabItems={vocabItems}/>)
    expect(wrapper).toMatchSnapshot();
})

test('should render message if no vocab items exist', () => {
    const wrapper = shallow(<VocabItemList vocabItems={[]}/>)
    expect(wrapper).toMatchSnapshot();
})