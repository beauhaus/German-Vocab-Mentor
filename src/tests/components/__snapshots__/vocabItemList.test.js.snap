// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render message if no vocab items exist 1`] = `
<div>
  <h1>
    VocabItemList
  </h1>
  <h2>
    (length: 
    <em>
      0
    </em>
    )
  </h2>
  <p>
    Sorry. I have no words.
  </p>
</div>
`;

exports[`should render vocabItems list with vocab items 1`] = `
<div>
  <h1>
    VocabItemList
  </h1>
  <h2>
    (length: 
    <em>
      3
    </em>
    )
  </h2>
  <h2>
    <VocabListItem
      createdAt={"1970-01-01T00:00:10.000Z"}
      difficulty="8"
      example_de="es weckt Erinnerungen"
      example_en="It awakens (brings back) memories"
      id={1}
      isShowing={true}
      key="1"
      note="memory note"
      word_de="Erinnerungen"
      word_en="Memories"
    />
    <VocabListItem
      createdAt={0}
      difficulty={5}
      example_de="Ich sehne mich nach Glühwein"
      example_en="I'm craving Glühwein"
      id={2}
      isShowing={true}
      key="2"
      note="wine note"
      word_de="Ich sehne mich nach"
      word_en="I long for/ I crave"
    />
    <VocabListItem
      createdAt={-10000}
      difficulty={8}
      example_de="Du hast meine Gedanken gelesen"
      example_en="You've read my thoughts"
      id={3}
      isShowing={false}
      key="3"
      note="thought note"
      word_de="Gedanken"
      word_en="Thought"
    />
  </h2>
</div>
`;
